src/main/java/org/betonquest/betonquest/database/MySQL.java

```diff
@@ -144,69 +159,98 @@ private void migration1(final Connection connection) throws SQLException {
-                    + "ADD PRIMARY KEY (profileID, objective) "
+                    + "MODIFY COLUMN instructions VARCHAR(2046) NOT NULL, "
+                    + "DROP PRIMARY KEY, "
+                    + "DROP COLUMN id, "
+                    + "ADD PRIMARY KEY (profileID, objective), "
                     + "ADD FOREIGN KEY (profileID) REFERENCES " + prefix + "profile (profileID) ON DELETE CASCADE");
-            statement.executeUpdate("DELETE FROM " + prefix + "points WHERE id NOT IN "
-                    + "(SELECT MIN(id) FROM " + prefix + "points GROUP BY playerID, category)");
+            deleteDuplicates(statement, "points", "category");
+            deleteOrphaned(statement, "points");
             statement.executeUpdate("ALTER TABLE " + prefix + "points "
                     + "CHANGE COLUMN playerID profileID CHAR(36) NOT NULL, "
-                    + "MODIFY COLUMN category VARCHAR(255) NOT NULL, " + "DROP PRIMARY KEY," + "DROP COLUMN id "
-                    + "ADD PRIMARY KEY (profileID, category) "
+                    + "MODIFY COLUMN category VARCHAR(255) NOT NULL, "
+                    + "DROP PRIMARY KEY, "
+                    + "DROP COLUMN id, "
+                    + "ADD PRIMARY KEY (profileID, category), "
                     + "ADD FOREIGN KEY (profileID) REFERENCES " + prefix + "profile (profileID) ON DELETE CASCADE");
-            statement.executeUpdate("DELETE FROM " + prefix + "tags WHERE id NOT IN "
-                    + "(SELECT MIN(id) FROM " + prefix + "tags GROUP BY playerID, tag)");
+            deleteDuplicates(statement, "tags", "tag");
+            deleteOrphaned(statement, "tags");
             statement.executeUpdate("ALTER TABLE " + prefix + "tags "
                     + "CHANGE COLUMN playerID profileID CHAR(36) NOT NULL, "
-                    + "MODIFY COLUMN tag VARCHAR(510) NOT NULL, " + "DROP PRIMARY KEY," + "DROP COLUMN id "
-                    + "ADD PRIMARY KEY (profileID, tag) "
+                    + "MODIFY COLUMN tag VARCHAR(510) NOT NULL, "
+                    + "DROP PRIMARY KEY, "
+                    + "DROP COLUMN id, "
+                    + "ADD PRIMARY KEY (profileID, tag), "
                     + "ADD FOREIGN KEY (profileID) REFERENCES " + prefix + "profile (profileID) ON DELETE CASCADE");
-            statement.executeUpdate("DELETE FROM " + prefix + "player WHERE id NOT IN "
-                    + "(SELECT MIN(id) FROM " + prefix + "player GROUP BY playerID)");
+            deleteDuplicates(statement, "player", null);
             statement.executeUpdate("ALTER TABLE " + prefix + "player "
-                    + "MODIFY COLUMN playerID CHAR(36) NOT NULL, " + "MODIFY COLUMN conversation VARCHAR(510), "
+                    + "MODIFY COLUMN playerID CHAR(36) NOT NULL, "
+                    + "MODIFY COLUMN conversation VARCHAR(510), "
                     + "ADD COLUMN active_profile CHAR(36) NOT NULL DEFAULT playerID AFTER playerID, "
```

Comment:
I put a space accidentaly when pasting from the terminal, it's still giving me the same error, I tested the syntax on a `test_table` in my sql and gives the same error:

<img width="1374" alt="CleanShot 2023-04-27 at 20 15 18@2x" src="https://user-images.githubusercontent.com/20805372/235016630-c5d5a6ca-016a-4231-b6bb-2a2488e2179e.png">

Then I saw this [answer](https://stackoverflow.com/a/71299479/13160500) on stackoverflow, and putting parenthesis on `PlayerId` in the `DEFAULT` statement works:

<img width="1085" alt="CleanShot 2023-04-27 at 20 30 26@2x" src="https://user-images.githubusercontent.com/20805372/235023424-6825c3c8-67ef-44f1-9cd0-ba6d4439e328.png">

I think that's the fix.

Reply:
I am a bit confused by this, because for me it worked flawless. What database (MariaDB,...) do you use? in what version?

Reply:
I now added the parenthesis, because i know it is no issue to have them there.

I really appreciate your review here. Thank you a lot. Let me know if you find any other issue. I think i will merge this PR soon. So it would be great to know that this has your approve ;)

For maybe faster interaction you can also write me in Discord.

Reply:
I'm using MySQL on ubuntu: `mysql  Ver 8.0.32-0ubuntu0.22.04.2 for Linux on x86_64 ((Ubuntu))`